### PR TITLE
fix: initial normalization

### PR DESCRIPTION
**Pull request by z0al**
Two things:

* Fixes the initial normalization logic. The existing logic had two issues:
  1. it throws `normalizeValue` is not a function error in some cases
  2. It triggers `onValueChanged` event which is not desired
* Says goodbye to `sanitizeIncomingSlateDoc` helper

### CODE REVIEW

**Review by cezarsmpio — APPROVED** [review]

File: cypress/integration/rich-text/RichTextEditor.spec.ts [hunk 1]
```diff
@@ -1566,13 +1566,59 @@ describe('Rich Text Editor', { viewportHeight: 2000 }, () => {
...
+        nodeType: 'document',
+        data: {},
+        content: [],
+      };
+
+      cy.setInitialValue(emptyDocument);
+
+      cy.reload();
+
+      // The field value in this case will still be untouched (i.e. invalid)
```
Comment (DanweDE) [inline cypress/integration/rich-text/RichTextEditor.spec.ts:1580]: Careful with words here as it's not "invalid" from the CMA's point of view but I guess you mean invalid in Slate terms? Perhaps

```suggestion
      // The field value in this case will still be untouched (i.e. unnormalized)
```

File: cypress/integration/rich-text/RichTextEditor.spec.ts [hunk 2]
```diff
@@ -1566,13 +1566,59 @@ describe('Rich Text Editor', { viewportHeight: 2000 }, () => {
...
   describe('invalid document structure', () => {
-    describe('normalizable errors', () => {
-      it('does not throw', () => {
-        cy.setInitialValue(invalidDocumentNormalizable);
-        cy.reload();
-        //check that editors content is what we expect (not a thrown error)
-        richText.expectSnapshotValue();
-      });
+    it('accepts document with no content', () => {
+      const emptyDocument = {
```
Comment (DanweDE) [inline cypress/integration/rich-text/RichTextEditor.spec.ts:1570]: Nit: I would prefer

```suggestion
      const docWithoutContent = {
```

as we explicitly state in [emptyDocument.ts](https://github.com/contentful/rich-text/blob/80315ef5130b79336336ba31de8a55e42bafe319/packages/rich-text-types/src/emptyDocument.ts#L6) what structure is considered to be an empty doc.

So until your structure gets normalized and receives an empty `text` node, it shouldn't really be considered empty.

**Review by DanweDE — APPROVED** [review]

File: cypress/integration/rich-text/RichTextEditor.spec.ts [hunk 3]
```diff
@@ -1566,13 +1566,59 @@ describe('Rich Text Editor', { viewportHeight: 2000 }, () => {
...
+      richText.editor.should('contain.text', 'cell #6');
+
+      // The field value in this case will still be untouched (i.e. invalid)
+      // since we won't trigger onChange.
+      richText.expectValue(invalidDocumentNormalizable);
+
+      // Initial normalization should not invoke onChange
+      cy.editorEvents()
+        .then((events) => events.filter((e) => e.type === 'onValueChanged'))
+        .should('deep.equal', []);
```
[the remark this answers is not in the document]
Comment (z0al) [reply cypress/integration/rich-text/RichTextEditor.spec.ts:1621]: I was just thinking about that. Will add the assertion now and merge

[the remark this answers is not in the document]
Comment (z0al) [reply cypress/integration/rich-text/RichTextEditor.spec.ts:1627]: Done

**Pull request merged**